File: HorseRacing.Contracts/Models/User/Requests/UpdateUser/UpdateUserRequest.cs
using HorseRacing.Contracts.Base.Requests;

namespace HorseRacing.Contracts.Models.User.Requests.UpdateUser
{
    public class UpdateUserRequest : BaseRequest<UpdateUserRequestDto>
    {
        public UpdateUserRequest() : base() { }
        public UpdateUserRequest(UpdateUserRequestDto data) : base(data) { }
    }
}

File: HorseRacing.Web/src/interfaces/api/contracts/model/user/requests/update-user/update-user-request.ts
import type { BaseRequest } from '~/interfaces/api/contracts/base/requests/base-request';
import type { UpdateUserRequestDto } from './update-user-request-dto';

export class UpdateUserRequest implements BaseRequest<UpdateUserRequestDto> {
    Data: UpdateUserRequestDto;

    constructor(data?: UpdateUserRequestDto) {
        this.Data = data || {
            UserId: '',
            FirstName: '',
            LastName: '',
            UserName: '',
            Phone: '',
            Email: '',
            Password: '',
        };
    }
}
